docs/content/preview/tutorials/AI/ai-llamaindex-openai.md

```diff
@@ -0,0 +1,250 @@
+
+Start a 3-node YugabyteDB cluster in Docker (or feel free to use another deployment option):
+
+```sh
+# NOTE: if the ~/yb_docker_data already exists on your machine, delete and re-create it
+mkdir ~/yb_docker_data
+
+docker network create custom-network
+
+docker run -d --name yugabytedb-node1 --net custom-network \
+    -p 15433:15433 -p 7001:7000 -p 9001:9000 -p 5433:5433 \
+    -v ~/yb_docker_data/node1:/home/yugabyte/yb_data --restart unless-stopped \
+    yugabytedb/yugabyte:2.20.1.0-b97 \
+    bin/yugabyted start \
+    --base_dir=/home/yugabyte/yb_data --background=false
+
+docker run -d --name yugabytedb-node2 --net custom-network \
+    -p 15434:15433 -p 7002:7000 -p 9002:9000 -p 5434:5433 \
+    -v ~/yb_docker_data/node2:/home/yugabyte/yb_data --restart unless-stopped \
+    yugabytedb/yugabyte:2.20.1.0-b97 \
+    bin/yugabyted start --join=yugabytedb-node1 \
+    --base_dir=/home/yugabyte/yb_data --background=false
+
+docker run -d --name yugabytedb-node3 --net custom-network \
+    -p 15435:15433 -p 7003:7000 -p 9003:9000 -p 5435:5433 \
+    -v ~/yb_docker_data/node3:/home/yugabyte/yb_data --restart unless-stopped \
+    yugabytedb/yugabyte:2.20.1.0-b97 \
+    bin/yugabyted start --join=yugabytedb-node1 \
+    --base_dir=/home/yugabyte/yb_data --background=false
+```
+
+The database connectivity settings are provided in the `{project_dir}/.env` file and do not need to be changed if you started the cluster with the preceding command.
+
+Navigate to the YugabyteDB UI to confirm that the database is up and running, at <http://127.0.0.1:15433>.
+
+## Load the Financial Schema and Seed Data
+
+This application requires a database table with financial information for companies in the S&P 500. This schema includes a `companies` table. It also creates a read-only user role to prevent any destructive actions while querying the database directly from LlamaIndex.
+
+1. Copy the schema to the first node's Docker container.
+    ```sh
+    docker cp {project_dir}/sql/schema_simplified.sql yugabytedb-node1:/home
+    ```   
+
+2. Copy the seed data file to the Docker container.
+    ```sh
+    docker cp {project_dir}/sql/data_simplified.sql yugabytedb-node1:/home
+    ```
+
+3. Execute the SQL files against the database.
+    ```sh
+     docker exec -it yugabytedb-node1 bin/ysqlsh -h yugabytedb-node1 -c '\i /home/schema_simplified.sql'
+     docker exec -it yugabytedb-node1 bin/ysqlsh -h yugabytedb-node1 -c '\i /home/data_simplified.sql'
+    ```
+
+## Start the Application
+
+This command-line application takes an input in natural language and returns a response from LlamaIndex.
+
+1. Start the server.
+
+```sh
+python3 index.py
+```
+
+```output
+What is your question?
+```
+
+2. Provide a relevant question. For instance:
+
+```
+What is your question? 
+
+Provide a detailed company history for the company with the highest marketcap.
+```
+
```

Comment:
I would add a sentence before the `output` saying something like "The AI agent will do X and Y and then respond the way below"